### PR TITLE
feat: refine mindmap

### PR DESCRIPTION
**Pull request by doouding**
### Changed
- Allow to drag mind map node within or between mind maps to change its layout.
- New mind map style
- Fixed some mind map bugs
- Remove unnecessary middlewares

### CODE REVIEW

**Comment by doouding** [comment]
* **#7173** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/7173?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @doouding and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/presets/src/fragments/copilot-panel/utils/markdown-utils.ts [hunk 1]
```diff
@@ -6,8 +6,12 @@ import { Job, type Slice } from '@blocksuite/store';
 
 export async function getMarkdownFromSlice(host: EditorHost, slice: Slice) {
   const job = new Job({ collection: host.std.doc.collection });
+  const snapshot = await job.sliceToSnapshot(slice);
```
Comment (Saul-Mirone) [inline packages/presets/src/fragments/copilot-panel/utils/markdown-utils.ts:9]:
```suggestion

```

File: packages/presets/src/fragments/copilot-panel/utils/markdown-utils.ts [hunk 2]
```diff
@@ -6,8 +6,12 @@ import { Job, type Slice } from '@blocksuite/store';
...
   const markdownAdapter = new MarkdownAdapter(job);
-  const markdown = await markdownAdapter.fromSlice(slice);
+  const markdown = await markdownAdapter.fromSliceSnapshot({
```
Comment (Saul-Mirone) [inline packages/presets/src/fragments/copilot-panel/utils/markdown-utils.ts:11]:
```suggestion
  const markdown = await markdownAdapter.fromSlice(slice)
```

File: packages/presets/src/fragments/copilot-panel/utils/markdown-utils.ts (hunk 1 above)
Comment (Saul-Mirone) [inline packages/presets/src/fragments/copilot-panel/utils/markdown-utils.ts:9]: Why do you need this change?

File: packages/presets/src/fragments/copilot-panel/chat/logic.ts [hunk 3]
```diff
@@ -625,8 +625,11 @@ export type ChatMessage =
 
 const docToMarkdown = async (doc: Doc) => {
   const job = new Job({ collection: doc.collection });
-  const markdownAdapter = new MarkdownAdapter(job);
-  const result = await markdownAdapter.fromDoc(doc);
+  const snapshot = await job.docToSnapshot(doc);
+  const result = await new MarkdownAdapter(job).fromDocSnapshot({
```
Comment (Saul-Mirone) [inline packages/presets/src/fragments/copilot-panel/chat/logic.ts:629]: Why do you need this change?

File: packages/presets/src/fragments/copilot-panel/utils/markdown-utils.ts (hunk 1 above)
Comment (doouding) [reply packages/presets/src/fragments/copilot-panel/utils/markdown-utils.ts:9]: These changes are introduced when resolving conflicts. Let me see if I can remove them.

**Review by Saul-Mirone — APPROVED** [review]

**Pull request merged**